gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiAuthorizationServiceImpl.java

```diff
@@ -313,6 +321,11 @@ public List<ApiCriteria> computeApiCriteriaForUser(
                 apiCriteriaList.add(queryToCriteria(executionContext, apiQuery).ids(userGroupApiIds).build());
             }
 
+            if (!isBlank(apiQuery.getCategory())) {
+                var categoryName = categoryService.findById(apiQuery.getCategory(), executionContext.getEnvironmentId()).getKey();
```

Comment:
why we had this for current implementation while building query
`builder.category(categoryService.findById(query.getCategory(), executionContext.getEnvironmentId()).getId())`
while in db we keep the name of the category, not id 
did you found the answer? Maybe there is a bug?

Reply:
In Apis table we keep not categories ids, but they keys, which are the same as category name

Reply:
I know, but I saw in code places that set api query category to id and not the name like you did